Core/Src/pdu.c

```diff
@@ -15,26 +16,75 @@
 
 #define SHUTDOWN_ADDR PCA_I2C_ADDR_3
 #define CTRL_ADDR     PCA_I2C_ADDR_2
-#define RTDS_DURATION 2500
+#define RTDS_DURATION 2500 /* ms at 1kHz tick rate */
 
 static osMutexAttr_t pdu_mutex_attributes;
 
-static void rtds_shutoff_cb(void *pv_params)
+static uint8_t sound_rtds(pdu_t *pdu)
+{
+	if (!pdu)
+		return -1;
+
+	osStatus_t stat = osMutexAcquire(pdu->mutex, MUTEX_TIMEOUT);
```

Comment:
I feel like mutex timeout should be 0 as a guarrantee we never will be awaiting this.

Reply:
This function is no longer called from a timer.